aten/src/ATen/native/cuda/ScanKernels.cpp

```diff
@@ -88,6 +88,9 @@ Tensor _logcumsumexp_cuda(const Tensor& self, int64_t dim) {
 }
 
 void cumsum_cuda_kernel(const Tensor& result, const Tensor& self, int64_t dim) {
+  // See Note [Writing Nondeterministic Operations]
+  // Issue reporting nondeterministic behavior: https://github.com/pytorch/pytorch/issues/75240
+  globalContext().alertNotDeterministic("cumsum_cuda_kernel");
```

Comment:
Hm, if we don't want to break the world (a lot of workloads enable deterministic flag and do cumsums) we'll have to limit it to floating datatypes only.

Reply:
Changed to only alert for floating and complex